src/coreclr/gc/vxsort/CMakeLists.txt

```diff
@@ -26,4 +26,4 @@ set (VXSORT_SOURCES
   do_vxsort.h
 )
 
-add_library(gc_vxsort STATIC ${VXSORT_SOURCES})
+add_library(gc_vxsort OBJECT ${VXSORT_SOURCES})
```

Comment:
I think it would be nice to use the same style, either this one in `src/coreclr/gc/unix/CMakeLists.txt` or the other way around.

Reply:
Do you mean to conditionally add gc_vxsort objects if android host?

Reply:
That or change static to object in gc/unix as well.

Reply:
I added gc_vxsort for the android host. I could also switch it from static to an object library, but I’m not sure the archive is actually being used.

Reply:
Do we need both, static and objects, for Android? If not, we can use an if-else block.

Reply:
BTW: vxsort is a ton of code for relatively small performance gain. We may want to disable it on mobile. It is implemented on x64 only for now, but arm64 implementation is in the works https://github.com/dotnet/runtime/pull/110692 .

Reply:
Personally, I'd prefer to switch from static to object library here instead of introducing the conditional object for Android. At least for `gc_pal` and `gc_vxsort`, since they are pretty isolated (used for coreclr and standalone gc) and I think the conditional object build makes it more complicated that just switching - as opposed to something like `coreclrminipal`, which would fit better as a separate change per https://github.com/dotnet/runtime/pull/114629#pullrequestreview-2803129889.

Reply:
I agree. I haven’t applied these changes to the other libraries in this PR, since I’m not sure whether they’re required as archives on other platforms.